### DISCUSSION TITLE
Some suggestions for MP lobby

### DISCUSSION THREAD

**Discussion by mp5tournament** [ppy/osu#29052] [Ideas]
I think some settings can be added, and special lobby for tournaments can be launched (they can be separated and placed side by side with ordinary MP lobby, or not separated and convert all the functions of tournaments lobby to settings). For example:

- Add optional referee slots, which do not occupy ordinary slots like stable and may also play maps (especially short spinner will cause the score to be non-0).
- Add optional settings such as no fail without nf mod / no team fail / nf no deduction / team's total score will include failed scores, to make the tournaments matches more convenient.
- Add optional automatic calculation of ez or other mod multiplier scores. This option does not affect the scores uploaded by the players themselves, but affects the team victory settlement.
- Lock some players' team and slot. This means that when a player enters the match lobby, he does not need to choose his own color and cannot enter the other team's slot.
- Some other functions that may help current common tournaments.
- (For non-tournament MP) The map selector and the host can become two roles, that is, the host can modify various settings, but the map selector can only select the map. The following two suggestions are based on the premise that the map selector and the host can be different.
- Add optional settings to limit the range of pictures (sr, length, etc.).
- Add optional settings to implement automatic rotation.

**Comment by peppy** [comment]
> Add optional referee slots, which do not occupy ordinary slots like stable and may also play maps (especially short spinner will cause the score to be non-0).

Is there a reason you can't use the spectate mode for this? That was the intention.
